compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -2904,14 +2904,35 @@ private boolean checkFillerValue(BUnionType type) {
         if (type.isNullable()) {
             return true;
         }
-        Iterator<BType> iterator = type.getMemberTypes().iterator();
+        Set<BType> memberTypes = new HashSet<>();
+        boolean hasFillerValue = false;
+        for (BType member : type.getMemberTypes()) {
+            if (member.tag == TypeTags.FINITE) {
+                Set<BType> uniqueValues = getUniqueValues((BFiniteType) member);
+                memberTypes.addAll(uniqueValues);
+            } else {
+                memberTypes.add(member);
+            }
+            if (!hasFillerValue && hasFillerValue(member)) {
```

Comment:
Where do we do the check too see if the filler value is part of this type?

e.g.,
```ballerina
const FOO = 1;
type Bar FOO | 2;

public function main() {
    Bar[2] b = [];
}
```

Does this result in an error?

Reply:
fixed thanks. 41ae36005fc63e09d9c63bcf6f14af7634f495f9